### ISSUE TITLE
rsync unexplained error transferring 600GB from Danforth to NCSA

### ISSUE THREAD

**Issue by dlebauer** [terraref/computing-pipeline#44]
I am trying to transfer ~600 GB from Danforth to Roger. I have set up SSH key pairs and use the following command:

```sh
nohup rsync -routivvvL apollo.ddpsc.org:/home/dlebauer/sorghum* /home/dlebauer/terraref/genomics/ &
```

But this fails with the following error (from `tail -n 5 /home/dlebauer/terraref/genomics/nohup.out` on Roger):

> calling match_sums /home/dlebauer/sorghum_accessions_data/SRR485949_2.fastq.gz
Killed by signal 1.
rsync error: unexplained error (code 255) at rsync.c(546) [generator=3.0.6]rsync error: received SIGINT, SIGTERM, or SIGHUP (code 20) at rsync.c(546) [receiver=3.0.6]

> _exit_cleanup(code=20, file=rsync.c, line=546): about to call exit(255)
_exit_cleanup(code=20, file=rsync.c, line=546): about to call exit(20)

@jdmaloney suggests two options:
1. this should be done on a data transfer node rather than the head node and
2. for large data transfers, we can use Globus

* [ ] @nfahlgren does the Apollo server have a Globus end point?
* [ ] @jdmaloney can you please give me access to a data transfer node?
* [ ] is there a reason for this error (perhaps a limit on head node use on roger?

**Comment by yanliu-chn** [comment]
We shall not transfer 600GB to $HOME/. Project dir or scratch space should be used. $HOME size is set to be < 10GB, by policy. I think the quota is not checked for $HOME at this moment. But might be an issue once quota is in place. Project space is in /projects/arpae/ .

**Comment by dlebauer** [comment]
@yanliu-chn ok. I am moving them now to /projects/arpae/genomics, once the local files are moved I will restart the remote rsync with 

```sh
nohup rsync -routivvvL apollo.ddpsc.org:/home/dlebauer/sorghum* /home/dlebauer/terraref/genomics/ &
```

**Comment by jdmaloney** [comment]
Hi David,
I will get you access to that node either tonight or tomorrow depending on time as I have a few things tonight.  Important note, I updated the /etc/motd on ROGER this afternoon.  The GPFS file system will be experiencing a planned outage tomorrow morning from 9am to approximately 10am CST as I do some necessary work on the home file system.  Not sure how long the rsync of the current data will take but I would hold of on starting the next transfer until my work is finished which I will email you directly when it is done.  Great part about rsync is it’ll just pick up where it leaves off, but just as a warning as you will see it get interrupted.

Best,

J.D. Maloney
Storage Engineer | Storage Enabling Technologies Group
National Center for Supercomputing Applications (NCSA)
University of Illinois Urbana-Champaign
1205 W. Clark St. Urbana, IL 61801

**Comment by yanliu-chn** [comment]
I didn’t have problem with rsync test. Did you try without nohup? -Yan

**Comment by dlebauer** [comment]
I am using nohup but the error occurs after a few hours. I've started it up again and will let you know.

**Comment by dlebauer** [comment]
> Great part about rsync is it’ll just pick up where it leaves off, but just as a warning as you will see it get interrupted.

since this is the case, I went ahead and restarted the rsync.

Using ` --progress --stats ` flags I see it is running at 3.7 MB/s, which should be ~14GB/h or ~ 45h for the 600GB transfer (which is 1/6 done).

**Comment by nfahlgren** [comment]
@dlebauer, setting up Globus is on my to-do list but it has not been set up yet.

**Comment by dlebauer** [comment]
Using the project directory `/projects/arpae` has not produced the issue - its been going for almost a day now whereas before it would stop after a few hours ... so I am closing this issue.

@jdmaloney feel free to kill these (PID 6018 and 6020) for maintenance you mentioned and let me know when I can re-start.

**Comment by jdmaloney** [comment]
Hi David,
File System was able to correct itself last night without a downtime which is better than we anticipated so no outage was taken today.  Still working on getting you access to that data mover node, will let you know when that is done.

Best,

J.D. Maloney
Storage Engineer | Storage Enabling Technologies Group
National Center for Supercomputing Applications (NCSA)
University of Illinois Urbana-Champaign
1205 W. Clark St. Urbana, IL 61801

**Comment by jdmaloney** [comment]
See if you can log on to cg-hm01 from the head node (cg-gpu01) that is the special node for transfers that we like to keep big transfer sessions on and has a dedicated 40GbE link, though it seems like we’re very much limited by the WAN connection of Danforth or something in between.

Best,

J.D. Maloney
Storage Engineer | Storage Enabling Technologies Group
National Center for Supercomputing Applications (NCSA)
University of Illinois Urbana-Champaign
1205 W. Clark St. Urbana, IL 61801

**Comment by dlebauer** [comment]
How do I log on to a specific node? ssh?

**Comment by jdmaloney** [comment]
Yep, ssh dlebauer@cg-hm01

Best,

J.D. Maloney

On Jan 6, 2016, at 11:00 PM, David LeBauer <notifications@github.com<mailto:notifications@github.com>> wrote:

How do I log on to a specific node? ssh?

**Comment by dlebauer** [comment]
@jdmaloney I can't log in. I get 

```sh
[terraref]$ ssh dlebauer@cg-hm01
Connection closed by 141.142.169.13
```

**Comment by jdmaloney** [comment]
Looks like things got blown away when all systems got rebuilt with some of the recent security patches, I’ve add you to cg-hm02…see if that works?

Best,

J.D. Maloney
Storage Engineer | Storage Enabling Technologies Group
National Center for Supercomputing Applications (NCSA)
University of Illinois Urbana-Champaign
1205 W. Clark St. Urbana, IL 61801
